Julia

JKRT opened the pull request:
@sjoelund 
Commit spam for now I hope that it is ok. I have started work on a testsuite it depends on MetaModelica.jl 

My idea was to divide it up into syntax tests and semantic tests with the same tests I wrote for my LLVM stuff 

(I am far from an expert in Julia so if you have some comments on it please leave them so I learn)

JKRT committed:
enhancement-5444: Now possible to escape while drawing lines and polygons

Commit:
[OMSI] Added test for algebraic loops

Added models with linear and non-linear loops to test default OMSIBase solver
for algebraic loops.

niklwors committed:
[OMSICpp] Add OMSI Cpp library and  omsicpp code target template

Simulation runtime for FMU/OMSU simulation in C++.

 - Added OMSICpp build
   - Use CMake to build SimultaionRuntime/OMSICpp
 - Added OMSICpp in Simulation Runtime
   - The code targets cpp and cpp fmu using this runtime also now
   - Using OMSIBase library for base functionalities shared with OMSIC runtime
   - Wrapper for FMI 2.0 ModelExchange functions
     - Functions for continuous simulation of FMU/OMSU
     - Functions for event simulation of FMU/OMSU
     - Getter and Setter functions for FMU/OMSU
     - Logging and some debugging functionalities
     - Initialization and deallocation of FMU/OMSU
   - OMSICpp Simulation runtime which supports omsu/fmu 2.0 ME and  the old
     cpp code traget systems
 - Documentation with Doxygen
   - Doxyfile not included

Co-authored-by: AnHeuermann <andreas.heuermann@fh-bielefeld.de>
Co-authored-by: wibraun <wbraun@fh-bielefeld.de>

sjoelund committed:
Prevent some memory leaks in KINSol

Related to ticket:5485. Note that free'ing the KINSol data makes
simulations time out so that part is not disabled.

adrpo committed:
- more NF API changes
- expose DynamicSelect expressions in annotations
- avoid parameter in annotation records to force scalarization, use nfAPI flag to force it
- handle DynamicSelect typing and evaluation
- support conversion from Boolean to Real (used in annotations in MSL, i.e. Boolean > 0.5) only if -d=nfAPI is on and issue a warning
- fix for ticket:5506, add test: testsuite/openmodelica/interactive-API/Ticket5506
- fix for ticket:5502, return just the first expression in DynamicSelect for all API if -d=nfAPIDynamicSelect flag is not given
- update testsuite/openmodelica/interactive-API/interactive_api_annotations

sjoelund committed:
Test Julia conversion

Commit:
Made homotopy initialization test independent of OS

perost committed:
[NF] Avoid evaluating external functions.

- Mark external function calls as non-structural, to avoid expressions
  containing such calls from being marked as structural when its not
  strictly necessary.

adrpo committed:
fix MSVC build
- handle MSVC that doesn't know how to handle array declarations with sizes set by local variables on the stack

adeas31 committed:
Documentation about DynamicSelect and state machine debugging

sjoelund committed:
Abort previous builds when PR has a new commit

JKRT committed:
RemovedOutCommentedCode: Removed code that was not used in Commands.cpp. They are still in the history so should be retrievable

JKRT committed:
ISSUE-5503: Focus set proper when changing the shape that we are activly drawing

JKRT committed:
ISSUE-5435: Now possible to reset OMEdit via the UI

ISSUE-5435: Changed to utf8. changed variables according to coding rules for OMEDit

ISSUE-5435: Added translation support for QT versions < 5.4

sjoelund committed:
Update ModelicaExternalC

This fixes ticket:5469.

adeas31 committed:
Always send utf-8 data from OMEdit to OMC

Use the wide char APIs for file handling

adeas31 committed:
Better reading of visual xml file

Fixes memory leaks.

JKRT committed:
ISSUE-5507: Prevents OMEdit from crashing when parsing text annotations. Fixed incorrect bound check

sjoelund committed:
Add Jenkins pipelines for slower jobs

adeas31 committed:
Use the Windows short path names to make MINGW happy

Windows creates are short path name for each folder that contains spaces or special characters. If we use the that MINGW can compile files from such paths.
Use the -w flag for make to print the current directory

sjoelund committed:
Use a mutex for nodes that share docker volumes

This avoids problems when checking out git repositories that are
cached using docker volumes.

sjoelund committed:
The clean target should clean difftool

sjoelund committed:
Actually clean difftool on make clean

adrpo committed:
- use NF for getComponents  API

adrpo committed:
[NF] minor change to remove duplication

sjoelund committed:
Fixes for parallel builds

sjoelund committed:
Add pipeline for cppruntime tests

sjoelund committed:
Fixes for latest sphinx version

adrpo committed:
more fixes for ticket:5507
- give defaults for 'extent' and 'textString' also in the Text record
other changes
- split execstat in NFApi for better performance info
- display the OMEdit log numbers with better formatting

JKRT committed:
Readded the missing files (#1)

Added the missing Julia files

sjoelund committed:
Add a pipeline for OSX

adeas31 committed:
Use the macros for unicode to utf-8 and back conversions

Added a `omc_file.h/c` which contains the file system operations like fopen, stat etc. Use the functions declared in this file instead of directly calling them.

perost committed:
[NF] Set structural parameters as final in the DAE

- Set structural parameters as final when creating the DAE structure,
  to indicate that they've been evaluated and shouldn't be modified.
- Disable the old way of doing this used by the OF when using the NF,
  to avoid unnecessary traversal of the DAE.

lochel committed:
Adjust minimal step size (#219)

sjoelund committed:
Add tests for alternative OS/compilers

niklwors committed:
Applied codestyle to omsicpp runtime

kabdelhak committed:
[BE,daeMode] add removed equations from shared

sjoelund committed:
Enable unit checking for newInst + checkModel

sjoelund committed:
Add test for checkModel performing unit check

adrpo committed:
add staging and pipeline commands for Windows/MinGW (optional)
- remove tabs
- put parentheses for the if condition
- escape $ in string -> \$
- escape \ -> \\
- shorten the name of the stage Windows/MinGW -> Win/MinGW
- use msys sh on Windows
- export OPENMODELICAHOME and OPENMODELICALIBRARY
- hardcode OMDEV for now, print PATH on Windows for debugging
- use bat instead of sh on Windows
- use cygpath to go from windows $WORKSPACE to msys directory
- more magic with quoting the Windows paths in the msys script
- use: set MSYS2_PATH_TYPE=inherit
- resurect runtimeCPPinstall target and make it point to OMSICPPinstall

sjoelund committed:
Use gmake instead of make

sjoelund committed:
Make it an option to test OSX

sjoelund committed:
Fix unit checking error in tested full model

Commit:
Fix OMSISolver build on macOS

  - Made reading of sundials version more robust in OMSI/solver/CMakeList.txt
  - Only use Sundials from 3rdParty because specific header are needed

sjoelund committed:
Use gmake instead of make

Forgot this part in 6da56e2.

adrpo committed:
update OMOptim submodule

sjoelund committed:
Do not require amd64 linux to run the milestone job

adrpo committed:
fix MSVC build after adding of omc_file.c|h
- fix the msvc part of omc_file.h

JKRT committed:
Possible to specify custom omc

sjoelund committed:
Use a label for MINGW/OSX builds in PR

By setting the label CI/Build OSX (or MINGW), Jenkins will now build
this PR also with the selected target(s). Useful if you do not have
access to a Mac machine and want to make sure you don't break it.

JKRT committed:
Functions now have typed return value. Deprecated the use of Real, use ModelicaReal and AbstractFloat instead

adrpo committed:
Fix OMSICpp build on MSVC
- change OMC_LIB to LIB_OMC inside the OMSICpp project and the makefiles
- properly propagate $LIB_OMC via cmake

adrpo committed:
add more deps for OMSI msvc

adrpo committed:
Fix a pch issue when building release

adrpo committed:
more fixes for building OMSICpp on MinGW
- link with shlwapi as is needed by fmil lib
- revert to OMC_LIB from LIB_OMC as is actually the libOpenModelicaCompiler in that case
- propagate LIB_OMC to more parts when building OMSICpp
- guard with IF(MSVC) installation of $TARGET_PDB_FILE

JKRT committed:
Added testsuite for MMToJulia. WIP for a s-s based on absyn

lochel committed:
Add license badge to README.md

perost committed:
[NF] Improve expandability check for if-equations.

sjoelund committed:
Skip building libraries on OSX check gmake

JKRT committed:
Added a utility function to fetch direction of an ElementItem

adrpo committed:
Revert "more fixes for building OMSICpp on MinGW"

This reverts commit 3ae6d2dc06abea8b8c6cac2c9b3867d876de5ac4.

Revert "Fix a pch issue when building release"

This reverts commit 7247254c0320c6096415d5669194402c470be8d6.

Revert "add more deps for OMSI msvc"

This reverts commit 9fd7ac7e08cc44524c9621cd797a5c229f235f20.

Revert "Fix OMSICpp build on MSVC"

This reverts commit 33e993f8f0e123efd631d29c782490939827e7aa.

Revert "Applied codestyle to omsicpp runtime"

This reverts commit dc1348bd5542e569a6affa243ffd64d2e1a6536b.

Revert "[OMSICpp] Add OMSI Cpp library and  omsicpp code target template"

This reverts commit 8633ec15ac094110f7810b99ff4ae12010f21801.

adrpo committed:
use withEnv for both MacOS and Win/MinGW pipelines
- this is so we do not share the env.PATH variable
- quote "git clean -ffdx" for the submodule command for Windows
  as otherwise you get error: unknown switch `f'
- do not put spaces in the MSYS_WORKSPACE as make/gcc cannot handle them later on
- quote ${WORKSPACE} when giving it to cygpath, echo the MSYS_WORKSPACE for debugging

JKRT committed:
Updated testsuite

JKRT committed:
WIP: Compiles. However output from Absyn.mo is wrong on very many levels

JKRT committed:
Updated .gitignore for previous input

JKRT committed:
Added a method to extract the TypeSpec from a given element item

JKRT committed:
Additional utility functions for Absyn -> Julia conversion

perost committed:
[NF] Fix fixed binding check for records.

JKRT committed:
Seems to translate very basic functions correctly

JKRT committed:
Uncommented chec

perost committed:
[NF] Improve handling of min/max of Booleans.

- Implement missing min/max_boolean_array functions in the runtime.
- Add min/max(arg) => arg optimization when arg is an array with a
  single element.

adeas31 committed:
Bumped OMSimulator

Fixed SEMLA compilation

kabdelhak committed:
[BE] added usage of flag -d=tearingdump and -d=tearingdumpV
  - extends -d=optdaedump to print residual/inner equations and
    tearing/inner variables instead of only printing indices

adrpo committed:
allow Win/MinGW stage to run on all nodes labeled windows

JKRT committed:
Added simple makefile to run tests

JKRT committed:
WIP: Makefile for tests. Updated codegen..

JKRT committed:
Now includes AbsynToJulia.tpl...

adrpo committed:
exit on error on the Win/MinGW pipeline

JKRT committed:
Updated tests

JKRT committed:
Added explicit return to home directory

sjoelund commented:
Seems OK so far from skimming the changes. New template AbsynToJulia and some changes to utility functions to make things work I suppose.

JKRT commented on OMCompiler/toJulia/lexer.jl:
```diff
@@ -15,10 +15,11 @@ keywordWithAction = [
 string   = re"\"([^\"\\x5c]|(\\x5c['\"?\\x5cabfnrtv]))*\""
 ident    = re"[_A-Za-z][_A-Za-z0-9]*|'([^'\\x5c]|(\\x5c.))+'"
 int      = re"[0-9]+"
+bool     = re "(tru|fals)e"
```
@sjoelund Is this correct? I am not sure.. my regexp skills are not exactly 100%

sjoelund replied:
Should be correct

JKRT commented on OMCompiler/toJulia/tokens.jl:
```diff
@@ -1,15 +1,19 @@
-  value::Real
+  value::ModelicaReal
+end
+struct BoolToken <: Token
```
@sjoelund  When I added this I thought it was necessary, but everything is  still more or less WIP

sjoelund replied:
Probably not necessary as it's just tokens

JKRT reviewed (commented):
Some minor comments

JKRT commented:
> 
> 
> Seems OK so far from skimming the changes. New template AbsynToJulia and some changes to utility functions to make things work I suppose.

More or less, what does not work (It is really quite many things that do not work) in the AbsynToJulia.tpl are Uniontypes, Absyn also surprises me a little bit by giving me Modelica records when I ought to receive MM_Records. Is this imagination or is it an issue with Absyn?

JKRT committed:
Yesterdays changes. Fixed some issues with MRV in MM to JL translation

JKRT commented on OMCompiler/toJulia/MMToJuliaTestSuite/syntaxCheck.jl:
```diff
@@ -0,0 +1,109 @@
+  cd(homeDirectory)
+end
+
+function syntaxCheck(omc)
```
This test currently does two things. It both tests the syntax and if AbsynToJulia.tpl translates without throwing errors. Currently, my algorithms test is failing. I plan to add a test for translating Graphviz at some point when ./Algorithm is translated correctly. 

When the syntax tests passes I will add semantic tests that tests these different functions

sjoelund commented:
Yes, Absyn does lack some of the uniontype things. It's stored just like it is in the source code (regular record and only inside the uniontype). Can probably be handled in some nice way for uniontypes (asserting that uniontype containing records are handled in a different way). The main difficulty might be for some of the packages that are actually uniontypes like NFExpression.mo.

arun3688 committed:
add more modelica proceedings 2019

JKRT committed:
Now compiles again?

lochel committed:
Update OMNotebook/.gitignore

AnHeuermann committed:
Add OMSI documentation (#242)

* added OMSI doxygen generation
* Renamed omsi -> OMSI and added target

perost committed:
[NF] Fix for ticket 5534.

- Always evaluate structural parameters in type attributes, to avoid
  scalarization issues.

kabdelhak committed:
[BE] cleanup -d=bltdump
 - more comprehensible output of MSSS and state selection
 - remove too verbose dumping of full system

Commit:
Added analytical jacobian for linear strong components

  - Added new flag LSanalyticJacobian to enable analytical jacobian for linear loops
    Defaults to false
  - Generate symbolic jacobian for linear systems when flag is enabled
  - Added testcase for linear solvers with analytical jacobian
  - Added testcase to dump sparsity pattern for linear problem

Commit:
Linear solvers: Check residuals and fixes for Lis

  - Fixed wrong index in analytic Jacobian lead index for Lis solver
  - Switched column and rows in getAnalytical Jacobian for Lis
  - Added check for all linear solver using jacobian matrices for residual vector equals null
  - Added missing linear solver for linear solver tests

adeas31 committed:
Use the short path names even for the installation path

JKRT committed:
Initial uniontype prototype. Things are now local where they should be for most cases

JKRT committed:
Updated alg tests

lochel committed:
Prefer tearing variables with start value for initialization (#256)

* Prefer tearing variables with start value for initialization
* Deactivate preOptModule removeEqualFunctionCalls
* Fix IndexReduction.addStateOrderFinder
* Disable two unstable optimization test cases
* Update expected output

kabdelhak committed:
[BE] fix bltdump (#274)

- removed dumping of stateorder in handler function to prevent
   failing during initialzation with NO_STATESELECTION
 - simulating with -b=bltdump should have always failed before

JKRT committed:
Graphviz.jl can now be translated and fed into Julia without compile errors

perost committed:
[NF] Compare names when type checking connections.

atrosinenko committed:
Fix completion for dot-qualified names

Probably due to regression, completions were generated for
Model|{ica,...} but not for Modelica.Bl|{ocks,...} because in the second
case, the completer machinery was generating just the [Blocks], not
[Modelica.Blocks], so Qt completer filtered it out on its own.

adeas31 committed:
ticket:5333 Use the whole suggested word instead of completing it

JKRT committed:
Now tests both Absyn and Graphviz

adeas31 committed:
Skip the call to diffModelicaFileListings if before and after strings are same

JKRT committed:
Better handling of explicit returns

adeas31 committed:
ticket:5541 Enable ctrl+home

joshbode committed:
bump OMOptim

adeas31 committed:
Preserve the geometry of OMC CLI Window

and SimulationOutputWidget

lochel committed:
Fix fmi2Reset

adeas31 committed:
ticket:5519 Fixed updateConnection API

Made it possible to set the annotation with and without $annotation.

lochel committed:
Add test case for fmi2Reset

arun3688 committed:
implement new extraction algorithm for dataReconciliation

adeas31 committed:
Don't quote environment variable otherwise short names won't work

adeas31 committed:
ticket:5490 Enable new instantiation and API by default

Prompt the user if the new instantiation fails and allow translation with the old.
Added two new APIs `enableNewInstantiation` and `disableNewInstantiation` which allows on/off for the new frontend instead of calling `clearCommandLineOptions` and `setCommandLineOptions`.

adrpo committed:
Fix ticket:5548
- split CevalScriptBackend.cevalInteractiveFunctions3 into two functions
- move some functions from Interactive to InteractiveUtil
- add test for calling getAvailableMatchingAlgorithms() without crashing

JKRT committed:
Updated AbsynToJulia.tpl

JKRT committed:
Added support for anonymous generated functions in the generator

adeas31 committed:
Preserve the SimulationDialog geometry correctly

Added a new tab for interactive simulation and moved some control to reduce the height of the SimulationDialog

kabdelhak committed:
[BE] resolveLoops update
  - eqCrossNodes triple loops can be detected now
  - detection of equal structure (e.g. arrays)

[fixup] testsuite

JKRT committed:
Decoupled utility functions and logic from absyn to make it standalone

adrpo committed:
fix boostrapping
- patch Absyn C sources
- fix QT api generation
- fix MatlabTranslator test
- fix testsuite

JKRT committed:
Updated comments in Absyn.mo 

The comment no longer make any references to functions since there is no longer logic in Absyn

kabdelhak committed:
[BE] fix stateSelect.never
  - force dummy selection of stateSelect.never variables
  - change adjacency matrix of states with stateSelect.never
    to SOLVABILITY_SOLVED() in all equations they appear on
    only one side (only for static state selection)
  - additional check for known variables in solvability checks

[fixup] testsuite update

JKRT committed:
Build bug resolved now builds...

JKRT committed:
All syntax tests now run

JKRT committed:
Now runs make allTests without throwing errors

adrpo committed:
partial fix for ticket:4608
- properly reset newInst flag to the previous value
- do not fail in getComponents on full type qualification, just return the same path

adeas31 committed:
Return the compilation error

Commit:
Dynamically allocated sparse structure and generic jacobian evaluation

  - Changed `SPARSE_PATTERN` to be dynamically allocated
    - Updated code generation
    - Updated C runtime solvers
  - Added generic function to evaluate jacobian for ida and dassl
  - Added jacobianSymbolical.c and jacobianSymbolical.h

JKRT committed:
Removed outdated test

adrpo committed:
fix ticket:5431
- proper handling of modelica:// URI resources for Windows
- fix handling of start values/bindings for strings in the generated FMUs

Commit:
Added some comments for DASSL C files

JKRT committed:
Updated tests

Commit:
Updated NPendulum test to work on Windows

adrpo committed:
make OPENMODELICAHOME have precedence over exe/so path
- libOpenModelicaCompiler can be used from other programs that
  may not be installed in the bin directory of OpenModelica,
  using the path from that exe is really wrong!

perost committed:
[NF] Don't trust global connector flags.

- Set the "has stream operators" flag when typing a stream connector,
  rather than trusting what SCodeUtil has set. SCodeUtil operates on
  all loaded models rather than only the used classes, and the flag
  might be overwritten by API calls before the NF is called.
- Clear the overconstrained connector flag at the start of
  instantiation, since the NF already sets it itself.

JKRT committed:
Added semantic tests 96/96 passed

JKRT committed:
Updated semantic check.jl

JKRT committed:
Added new tests for algorithms

JKRT committed:
Updated compiler sources

JKRT committed:
Up

JKRT committed:
Additional bugfixes

JKRT committed:
No comma less head problem

JKRT committed:
Absyn now seem to translate without errors. Seems to work. More testing neeeded

JKRT committed:
Removed some unnecessary code

adrpo committed:
fix ticket:5431
- do NOT, i repeat, do NOT reset memory_pools->used in pool_free_extra_list
  as this will basically remove all the allocated stuff from the current
  memory pool

JKRT committed:
Added current Absyn to SCode output

JKRT committed:
Now handles input-output variables

adrpo committed:
better fix for ticket:5431
- check if URI has a drive letter, not if system is Windows as
  one could send the source code FMU to somebody who will compile
  it on Linux (and the URI will be in windows format)

adeas31 committed:
ticket:5554 Don't stretch the library icon

adeas31 committed:
ticket:5566 Load the animation window

even if the `Switch to plotting perspective after simulation` is unchecked.

JKRT committed:
Added SCode and SCodeUtil to the parse test. Currently both fail to parse. Should be two failures now

adeas31 committed:
Fixes ticket:2166 & ticket:5561

Show values of all the curves at a point.

kabdelhak committed:
[BE] fix computation of array record sizes
  - ticket 4611

JKRT committed:
The MM to Julia transpiler is now able to generate code for most of the frontend and julia is able to parse it :)

casella committed:
Added saveTotalFile documentation to NF

casella committed:
Added documentation of saveTotalFile to old FE

adrpo committed:
[NF] fix ticket:5577
- for an FMU build, evaluate OpenModelica_uriToFilename in NFCeval to OpenModelica_fmuLoadResource same as the old frontend does

JKRT committed:
Partial functions now work properly when translating from Absyn. Sadly comments will be lost to history

adeas31 committed:
Improved the error message of multiple top level entities

Tell the user that its not possible to load a model saved with `saveTotalModel`.

lochel committed:
Add doc/SimulationRuntime/OMSI/.gitignore

niklwors committed:
[cppRuntime] fix for ModelicaExternal C missing objects, fix for misssing status return of omc c-api init function, deactivated threads in c-api test

adrpo committed:
fix ticket:5571
- give default BitMap(imageSource = "")
- add test, update test

adrpo committed:
fix ticket:5565
- add test Ticket5565.mos
- quiet down the graphical API noise (only activated with -d=nfAPINoise)
- update tests

kabdelhak committed:
[FE] Expand structural annotations
  + InlineAfterIndexReduction
  + GenerateEvents

JKRT committed:
Now tests builtin sort

JKRT committed:
Able to run MetaModelica code correctly. Can now parse most of the frontend

JKRT committed:
removed old file

JKRT committed:
Updated copyright to current year (#325)

kabdelhak committed:
[BE] update InlineAfterIndexReduction
 - inline function calls with InlineAfterIndexReduction only for matching
 - ticket 5170
 - fixup testuite

JKRT committed:
Removed unused imports

These imports are not needed

JKRT committed:
Update AbsynUtil.mo

Readded dump

JKRT committed:
Added support for as operations

vwaurich committed:
add complete AutoDeskColorMap

lochel committed:
Improve Error.STRUCT_SINGULAR_SYSTEM

lochel committed:
Avoid printing trailing whitespaces

perost committed:
Handle fully qualified reduction expressions.

JKRT committed:
Fixes to the templates

JKRT committed:
Fixed missspelling

JKRT committed:
Can now successfully compile AbsynUtil (Some small changes were needed)

JKRT committed:
All necessary modules for AbsynToSCode can now be compiled into naitve code without errors (Warnings however)

JKRT committed:
Did I mess up OMEdit somehow?

adeas31 committed:
ticket:5537 Handle multiple parameter values in a TextAnnotation

Display the unit of the parameter if available. The unit is only shown if parameter has some value.

adeas31 committed:
ticket:5588 Avoid crash when deleting a connection without a component

adeas31 committed:
Information message about the usage of global translation flags

adeas31 committed:
Flush the logs if --Debug=true is set

kabdelhak committed:
[BE] consider protected toplevel inputs as unknown
- ticket #5591

perost committed:
[NF] Fix function derivative mapping.

- Add lower order derivatives to the function derivative mappings like
  the old frontend does.

adeas31 committed:
Delete the associated connections of unknown component

JKRT committed:
We can now handle MetaModelica style import. Added dependency to ExportAll.jl

JKRT committed:
Packages and scaffolding to Absyn -> Graphviz transformation

JKRT committed:
Fixed name imports

JKRT committed:
Now use IArrray instead of array

JKRT committed:
More modules

adrpo committed:
extra fixes to ticket:5565, more shush

adeas31 committed:
ticket:5441 Show the relevant toolbars based on the active model type

lochel committed:
Reactivate removeEqualFunctionCalls

lochel committed:
Make optdaedump printing the original system

JKRT committed:
Updated tests for Absyn

JKRT committed:
Missing files also

JKRT committed:
Updated tst match semantics

lochel committed:
Expected output

lochel committed:
Disable inlineArray1.mos

JKRT committed:
wip

JKRT committed:
WIP

JKRT committed:
Added support for dumping syntax tree. Old way is via dump did not work :(

JKRT committed:
Updated string gen

JKRT committed:
OMC seems to be able to generate Absyn for Julia

JKRT committed:
Updated tests

JKRT committed:
added DAE interface

Commit:
Build OMSI under OSX

  - added install_name_tool -change ... in makefile

adrpo committed:
more fixes for ticket:5565

JKRT committed:
We now handle match on fields where all elements are wildcards. Runtime is updated accordingly

sjoelund committed:
Update 3rdParty for dgesv fix including math.h

This resolves some issues with embedded systems using a special
`log()` function rather than the one assumed by f2c.

sjoelund committed:
Add a sanity check that the build folder can be moved

`make install` assumes that `omc` will work even if the installation
directory is moved. This should check that the rpath works fine; it will
not check if files are linked directly against temporary files in the
source directories though.

Changed implementation of getting the installation directory on
Linux by using dladdr like we do on OSX.

sjoelund committed:
Use __linux__ for C++11 compatibility

gcc -std=c++11 does not define linux on Linux OSes.

sjoelund committed:
Add a script to look for mutually dependent packages

sjoelund committed:
Give additional diagnostics if the omhome is bad

sjoelund committed:
Check to not override buffer in Sphinx help text

sjoelund committed:
Refactor SCode/SCodeUtil to become more independent

The packages no longer import any packages that use SCode/SCodeUtil
again.

sjoelund committed:
Moved around SCodeUtil

SCodeUtil was renamed AbsynToSCode and functions from SCode moved
to a new SCodeUtil package.

JKRT committed:
Ut

JKRT committed:
Alot of small fixes here and there...

sjoelund committed:
Split MetaUtil into SCode and front-end parts

sjoelund committed:
Add warning for mixed position/named matching

JKRT committed:
Changes after rebase

sjoelund committed:
Refactoring away mixed positional/named matching

JKRT committed:
wip

JKRT committed:
Updated mos

JKRT committed:
Made implicit import to Absyn explicit

JKRT committed:
Return type now in the right order

JKRT committed:
Updated creation scripts

JKRT committed:
Added ra script

JKRT committed:
Updated references

JKRT committed:
m ts

JKRT committed:
merge

JKRT committed:
Various fixes

JKRT committed:
to Martin

JKRT committed:
Update to s2s

JKRT committed:
Added split for FFT

sjoelund committed:
Split the Absyn .jl dump into multiple lines

sjoelund committed:
Fix list output
